lib/elixir/lib/string.ex

```diff
@@ -1779,6 +1779,22 @@ defmodule String do
   @doc ~S"""
   Checks whether `string` contains only valid characters.
 
+  `algorithm` may be `:default` or `:fast_ascii`. Both algorithms are equivalent from a validation
+  perspective (they will always produce the same output), but `:fast_ascii` can yield signifcant 
+  performance benefits in specific scenarios.
+
+  If all of the following conditions are true, you may want to experiment with the `:fast_ascii`
+  algorithm to see if it yields performance benefits in your specific scenario:
+
+  * You are running Erlang/OTP 26 or newer on a 64 bit platform
+  * You expect most of your strings to be longer than ~64 bytes
+  * You expect most of your strings to contain mostly ASCII codepoints
+
+  Note that the `:fast_ascii` algorithm does not affect correctness, you can expect the output of
+  `String.valid?/2` to be the same regardless of algorithm. The only difference to be expected is
+  one of performance, which can be expected to improve roughly quadratically in string length
```

Comment:
> improve roughly quadratically

I'm struggling a bit to understand this one, given that both algorithms are linear in string length.
I would have assumed the ratio to be capped by a maximum constant value?
Sorry if I misunderstood.

Reply:
That's a great point! I messed up here by taking an engineering approach ('the graph's fitting curve is quadratic!') rather than a first principles approach (in which both approaches are obviously linear by inspection, as you correctly point out). 

My error was basically failing 'chart literacy 101'. Here's the chart that I based my conclusion on:

<img width="738" alt="image" src="https://user-images.githubusercontent.com/79646/214906291-e0dc4a1e-c799-4b42-99ea-b3f89e355d0f.png">

Looks quadratic, gets fit well by a quadratic trend line, so it must be quadratic. That's as far as my thought process went. But look at the x axis! It's not linear! If I graph the same data as a proper scatter plot (ie: on a linear x axis), we get:

<img width="753" alt="image" src="https://user-images.githubusercontent.com/79646/214907792-14957127-6ed3-493e-b7bf-bc507fc42127.png">

which fits linear-ish (despite visually not fitting well, the R value is pretty strong).

So yeah. The improvement is actually linear as you suspect. 

I'll update the docs to reflect this. Good catch!

Reply:
Thank you for clarifying :heart:
Based on theory and the new graph, I wonder if we won't see a glass ceiling appear if we add a couple order magnitudes on string length?